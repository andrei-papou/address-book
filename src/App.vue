<template>

    <div id="app">
        <router-view></router-view>
    </div>

</template>

<script>

import { mapGetters, mapActions } from 'vuex';
import TopBar from '@/components/TopBar';

export default {
    name: 'app',
    components: {
        TopBar
    }
}

</script>

<style lang="scss">

@import './assets/scss/main.scss';

</style>
