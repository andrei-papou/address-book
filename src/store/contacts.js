import moment from 'moment';
import Vue from 'vue';
import contactsApi from '@/api/contacts';
import {
	CONTACTS_LIST,
	CONTACTS_EDIT,
	CONTACTS_CREATE,
} from './mutation-types';


const state = {
	all: [],
};

export const actions = {
	getContacts({ commit }) {
		return contactsApi.getContactList()
			.then(response => commit(CONTACTS_LIST, response))
			.catch(error => console.error(error));
	},

	createContact({ commit }, contactData) {
		return contactsApi.createContact(contactData)
			.then(() => commit(CONTACTS_CREATE, contactData))
			.catch(err => console.log(err));
	},

	updateContact({ commit }, updatedObj) {
		commit(CONTACTS_EDIT, updatedObj);
	},
};

export const mutations = {
	[CONTACTS_LIST](state, contacts) {
		contacts.forEach(c => c.birthday = moment(c.birthday, 'YYYY-MM-DD').format('MM/DD/YYYY'));
		state.all = contacts;
	},

	[CONTACTS_CREATE](state, newObj) {
		state.all.push(newObj);
	},

	[CONTACTS_EDIT](state, updatedObj) {
		const index = state.all.findIndex(c => c.name === updatedObj.name);
		Vue.set(state.all, index, Object.assign({}, state.all[index], updatedObj));
	}
};

const getters = {
	allContacts: state => state.all,
	getContactByName: state => name => state.all.find(c => c.name === name),
};

export default {
	state,
	actions,
	mutations,
	getters
};
