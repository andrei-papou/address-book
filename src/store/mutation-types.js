export const CONTACTS_LIST = 'contacts-list';
export const CONTACTS_CREATE = 'contacts-create';
export const CONTACTS_EDIT = 'contacts-edit';
