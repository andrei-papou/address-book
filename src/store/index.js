import Vuex from 'vuex';
import Vue from 'vue';
import contacts from './contacts';


Vue.use(Vuex);

export default new Vuex.Store({
	modules: {
		contacts
	}
});
