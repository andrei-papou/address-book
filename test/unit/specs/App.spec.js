import Vue from 'vue';
import App from '@/App';
import store from '@/store';
import router from '@/routes';
import { mockFetch } from '../mock.js';

describe('App.vue', () => {

	let vm;
	const Constructor = Vue.extend(App);

	beforeEach(() => {
		mockFetch({ ok: true, json: () => Promise.resolve([]) });
		vm = new Constructor({ store, router }).$mount();
	});

	it('should render correct dom elements', () => {
		expect(vm.$el).toMatchSnapshot();
	});

});
