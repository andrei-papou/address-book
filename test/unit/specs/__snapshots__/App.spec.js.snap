// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should render correct dom elements 1`] = `
<div
  id="app"
>
  <!---->
</div>
`;
